### PR TITLE
Fix the Gc.set function for OCaml versions between 4.08 and 4.11.

### PR DESCRIPTION
**Pull request by ppedrot**
This patch is based on a commit by @mroch to fix the same issue with Flow (https://github.com/noppa/flow/commit/81e84471a53693fd4ab062f49e5a3f0afe11f510). I think it is trivial enough to not qualify for licensing matters, but let us still attribute it to its author by this comment.

Funnily enough the PR requires OCaml >= 4.08 to compile, so it will have to be merged right after the version bump.

Depends on:
- #15947

### CODE REVIEW

File: sysinit/coqinit.ml [hunk 1]
```diff
@@ -10,6 +10,42 @@
...
+      {
+        control with
+        custom_major_ratio = custom_major_ratio lsr 1;
+        custom_minor_ratio = custom_minor_ratio lsr 1;
+        custom_minor_max_size = custom_minor_max_size lsr 1;
+      }
+    in
+    Gc.set control
+  in
+  let v =
```
Comment (mroch) [inline sysinit/coqinit.ml:29]: you might want to use `Coq_config.caml_version_nums` like in `init_gc`

Comment (ppedrot) [reply sysinit/coqinit.ml:29]: Right, thanks.

**Comment by ppedrot** [comment]
This is now ready to merge (fast).

**Review by gares — APPROVED** [review]

**Comment by gares** [comment]
@coqbot merge now

**Comment by SkySkimmer** [comment]
Postponed due to needing newer ocaml

**Pull request merged**